render/cross_section/cross_section_rendering_engine_4d.cpp

```diff
@@ -0,0 +1,133 @@
+#include "cross_section_rendering_engine_4d.h"
+
+#include "../../math/transform_4d.h"
+#include "../../model/mesh_instance_4d.h"
+#include "../../nodes/camera_4d.h"
+
+#if GDEXTENSION
+#include <godot_cpp/classes/material.h>
+#include <godot_cpp/classes/world_3d.h>
+#include <godot_cpp/classes/rendering_server.hpp>
+#include <godot_cpp/classes/viewport.hpp>
+#include <godot_cpp/core/error_macros.hpp">
+#include <godot_cpp/core/variant/typed_array.hpp"
+#include <godot_cpp/core/variant/variant.hpp>
+#elif GODOT_MODULE
+#include "core/error/error_macros.h"
+#include "core/variant/array.h"
+#include "core/variant/variant.h"
+#include "scene/main/viewport.h"
+#include "scene/resources/3d/world_3d.h"
+#include "scene/resources/material.h"
+#include "servers/rendering_server.h"
+#endif
+
+RID create_instance(const Viewport &viewport) {
+	ERR_FAIL_NULL_V(RenderingServer::get_singleton(), RID());
+	RID instance = RS::get_singleton()->instance_create();
```

Comment:
It seems that GDExtension does not define `RS`, so we either need to define that manually, or change all instances of `RS` to `RenderingServer`, which I'd argue is more readable anyway.

```suggestion
	RID instance = RenderingServer::get_singleton()->instance_create();
```